File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    cnt = (brown - 4) // 2

    yellow_div = [i for i in range(1, yellow + 1) if yellow % i == 0]

    for i in yellow_div:
        j = yellow // i

        w, h = max(i, j), min(i, j)
        if w * h == yellow and w + h == cnt:
            answer.append(w + 2)
            answer.append(h + 2)
            break;

    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        for (int i = 1; i <= yellow; i++) {
            if (yellow % i == 0) {
                int a = yellow / i;
                int b = i;
                if ((a+2) * (b+2) == brown+yellow) {
                    return new int[]{a+2, b+2};
                }
            }
        }
        return answer;
    }
}